docs/source/developer_guides/model_merging.md

```diff
@@ -0,0 +1,178 @@
+print(tokenizer.decode(outputs[0]))
+```
+
+</hfoption>
+<hfoption id="ad copy">
+
+```py
+messages = [
+    {"role": "system", "content": "Create a text ad given the following product and description."},
+    {"role": "user", "content": "Product: Sony PS5 PlayStation Console\nDescription: The PS5 console unleashes new gaming possibilities that you never anticipated."},
+]
+text = tokenizer.apply_chat_template(messages, add_generation_prompt=True, tokenize=False)
+inputs = tokenizer(text, return_tensors="pt")
+inputs = {k: v.to("cuda") for k, v in inputs.items()}
+outputs = model.generate(**inputs, max_new_tokens=128, do_sample=True, top_p=0.95, temperature=0.2, repetition_penalty=1.2, eos_token_id=tokenizer.eos_token_id)
+print(tokenizer.decode(outputs[0]))
+```
+
+</hfoption>
+<hfoption id="SQL">
+
+```py
+text = """Table: 2-11365528-2
+Columns: ['Team', 'Head Coach', 'President', 'Home Ground', 'Location']
+Natural Query: Who is the Head Coach of the team whose President is Mario Volarevic?
+SQL Query:"""
+
+inputs = tokenizer(text, return_tensors="pt")
+inputs = {k: v.to("cuda") for k, v in inputs.items()}
+outputs = model.generate(**inputs, max_new_tokens=64, repetition_penalty=1.1, eos_token_id=tokenizer("</s>").input_ids[-1])
+print(tokenizer.decode(outputs[0]))
+```
+
+</hfoption>
+</hfoptions>
+
+## DARE
+
+The DARE method uses the [`~utils.random_pruning`] approach to randomly drop parameters, and only preserve a percentage of the parameters set in the `density` parameter. The remaining tensors are rescaled to keep the expected output unchanged.
+
+With PEFT, DARE is enabled by setting `combination_type="dare_ties"` and setting `density` to a value of the weights to keep from the individual models.
+
+> [!TIP]
+> DARE is a super useful method for preparing models for merging which means it can be combined with other methods like TIES, `linear` (a weighted average of the task tensors) or `svd` (calculated from the *delta weights*, the model parameters before and after finetuning) or a combination of all of the above like `dare_ties_svd`.
+
+Let's merge three diffusion models to generate a variety of images in different styles using only one model. The models you'll use are (feel free to choose your own): [nerijs/pixel-art-xl](https://huggingface.co/nerijs/pixel-art-xl) and [ostris/super-cereal-sdxl-lora](https://huggingface.co/ostris/super-cereal-sdxl-lora).
+
+Load the base model and then use the [`~diffusers.loaders.StableDiffusionXLLoraLoaderMixin.load_lora_weights`] method to load and assign each adapter a name:
+
+```py
+from diffusers import StableDiffusionXLPipeline, AutoencoderKL
+import torch
+
+vae = AutoencoderKL.from_pretrained(
+        "madebyollin/sdxl-vae-fp16-fix",
+)
+pipeline = StableDiffusionXLPipeline.from_pretrained(
+    "stabilityai/stable-diffusion-xl-base-1.0",
+    vae=vae,
+    torch_dtype=torch.float16,
+).to("cuda")
+
+pipeline.load_lora_weights(
```

Comment:
@pacman100 is this possible or do we need to import the `load_combine_checkpoint` method from `train_dreambooth_peft` as shown in this [notebook](https://github.com/pacman100/peft-dreambooth-ui/blob/main/lora_merging.ipynb)?

Reply:
Hello, for Diffusers, the merging methods can't be used as they follow a ckpt format that is different from the one I use and as Diffusers use low-level API of PEFT, it doesn't have access to the `add_weighted_adapter` method. The repo that I have uses bespoke ckpt format and uses normal PEFT API to get access to the `add_weighted_adapter` method.

Reply:
Ah I see! Then maybe we shouldn't show an example of TIES/DARE model merging with Diffusers yet then since it's not supported and `peft_dreambooth_ui` is still in development. WDYT @BenjaminBossan?

Reply:
Yes, I agree, it would be confusing otherwise.